tests/utils_tests/test_datastructures.py

```diff
@@ -62,31 +94,49 @@ def test_appendlist(self):
 
     def test_copy(self):
         for copy_func in [copy.copy, lambda d: d.copy()]:
-            d1 = MultiValueDict({
-                "developers": ["Carl", "Fred"]
-            })
-            self.assertEqual(d1["developers"], "Fred")
-            d2 = copy_func(d1)
-            d2.update({"developers": "Groucho"})
-            self.assertEqual(d2["developers"], "Groucho")
-            self.assertEqual(d1["developers"], "Fred")
-
-            d1 = MultiValueDict({
-                "key": [[]]
-            })
-            self.assertEqual(d1["key"], [])
-            d2 = copy_func(d1)
-            d2["key"].append("Penguin")
-            self.assertEqual(d1["key"], ["Penguin"])
-            self.assertEqual(d2["key"], ["Penguin"])
+            with self.subTest(copy_func):
+                d1 = MultiValueDict({
+                    "developers": ["Carl", "Fred"]
+                })
+                self.assertEqual(d1["developers"], "Fred")
+                d2 = copy_func(d1)
+                d2.update({"developers": "Groucho"})
+                self.assertEqual(d2["developers"], "Groucho")
+                self.assertEqual(d1["developers"], "Fred")
+
+                d1 = MultiValueDict({
+                    "key": [[]]
+                })
+                self.assertEqual(d1["key"], [])
+                d2 = copy_func(d1)
+                d2["key"].append("Penguin")
+                self.assertEqual(d1["key"], ["Penguin"])
+                self.assertEqual(d2["key"], ["Penguin"])
+
+    def test_deepcopy(self):
+        d1 = MultiValueDict({'a': [[123]]})
+        d2 = copy.copy(d1)
+        d3 = copy.deepcopy(d1)
+        self.assertIs(d1['a'], d2['a'])
+        self.assertIsNot(d1['a'], d3['a'])
+
+    def test_getstate(self):
+        x = MultiValueDict({'a': ['1', '2'], 'b': ['3']})
+        self.assertEqual(x.__getstate__(), {'_data': {'a': ['1', '2'], 'b': ['3']}})
+
+    def test_setstate(self):
+        d1 = MultiValueDict({'a': ['1', '2'], 'b': ['3']})
+        d2 = MultiValueDict()
+        d2.__setstate__({'_data': {'a': ['1', '2'], 'b': ['3']}})
+        self.assertEqual(d1, d2)
```

Comment:
I think these are too internal, I would rather check that `MultiValueDict` is pickleable:
```python
pickle.loads(pickle.dumps(...))
```